File: Rust/algo/src/sort/bubble.rs
pub fn bubble_sort<T:Ord + Copy>(arr: &mut [T]) {
    for _ in 0..arr.len() {
        let mut swapped = false;
        for j in 0..arr.len()-1 {
            if arr[j] > arr[j+1] {
                (arr[j], arr[j+1]) = (arr[j+1], arr[j]);
                swapped = true;
            }
        }
        if swapped == false {
            break;
        }
    }
}

File: C/SORT/bubble.c
void bubble_sort(int arr[], int n)
{
    int swap;
    for(int j=0; j<n; j++){
        for(int i=0; i<n-1-j; i++){
            if(arr[i] > arr[i+1]){
                swap = arr[i]; 
                arr[i] = arr[i+1]; 
                arr[i+1] = swap;
            }
        }
    }
}
